version: "3.9"

services:
  app:
    container_name: app_api_py_app
    build: .
    env_file:
      - .env
    command: bash -c "
      uvicorn src.main:app
      --host 0.0.0.0 --port 8080
      --reload --log-level debug
      "
    volumes:
      - .:/home/code
    ports:
      - "8080:8080"
    depends_on:
      - db

  db:
    container_name: app_api_py_db
    build:
      context: ./db
      dockerfile: Dockerfile
    volumes:
      - app_api_py_postgresql_data:/var/lib/postgresql/data
    env_file:
      - .env
    ports:
      - 5432:5432
    environment:
      - POSTGRES_PASSWORD=${DB_PASSWORD}
      - POSTGRES_USER=${DB_USER}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -d $DB_NAME -U $DB_USER"]
      interval: 5s
      timeout: 5s
      retries: 5

  redis:
    image: redis:latest
    container_name: app_api_py_redis_redis
    ports:
      - "6379:6379"
    env_file:
      - .env
    entrypoint: redis-server --appendonly yes

volumes:
  app_api_py_postgresql_data: {}
